yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,740 @@
+            'manifest_url': m3u8_url,
+            'thumbnail': url_or_none(metadata.get('thumbnail')),
+            'description': str_or_none(metadata.get('description')),
+            'like_count': int_or_none(metadata.get('likeCount')),
+            'dislike_count': int_or_none(metadata.get('dislikeCount')),
+            'creator': str_or_none(traverse_obj(metadata, ('creator', 'name'))),
+            'channel': str_or_none(traverse_obj(metadata, ('creator', 'name'))),
+            'channel_id': traverse_obj(metadata, ('creator', 'id')),
+            'uploader_id': traverse_obj(metadata, ('creator', 'id')),
+            'channel_url': try_get(metadata, lambda x: url_or_none(self._SINGLE_VIDEO_BASE_WEB_URL + traverse_obj(x, ('creator', 'username')))),
+            'availability': availability,
+            'tags': metadata.get('tags', []),
+            'live_status': 'was_live' if (stream_scheduled_for is None) and (stream_ended_at is not None) else
+                           'is_live' if stream_scheduled_for is None else  # stream_scheduled_for=stream_ended_at=None
+                           'is_upcoming' if stream_ended_at is None else   # stream_scheduled_for is not None
+                           None,  # Both stream_scheduled_for and stream_ended_at are not None: inconsistent meta data.
+            # Remove the 'False and' part when Rokfin corrects the 'stoppedAt' field:
+            'duration': (stream_ended_at - stream_started_at).total_seconds() if False and stream_started_at and stream_ended_at else None,
+            'timestamp': release_timestamp,
+            'release_timestamp': release_timestamp,
+            'formats': frmts or [],
+            'subtitles': subs or {},
+            '__post_extractor': self.extract_comments(video_id=video_id)
+        }
+
+    def _get_comments(self, video_id):
+        raise ExtractorError(msg='downloading stream chat is unsupported', expected=True)
+
+
+class RokfinPlaylistIE(RokfinIE):
+    def _get_video_data(self, json_data, video_base_url):
+        def real_get_video_data(content):
+            media_type = content.get('mediaType')
+            fn = try_get(media_type, lambda y: {
+                'video': lambda x: {'id': x['id'], 'url': f'{video_base_url}post/{x["id"]}'},
+                'audio': lambda x: {'id': x['id'], 'url': f'{video_base_url}post/{x["id"]}'},
+                'stream': lambda x: {'id': x['mediaId'], 'url': f'{video_base_url}stream/{x["mediaId"]}'},
+                'dead_stream': lambda x: {'id': x['mediaId'], 'url': f'{video_base_url}stream/{x["mediaId"]}'},
+                'stack': lambda x: {'id': x['mediaId'], 'url': f'{video_base_url}stack/{x["mediaId"]}'},
+                'article': lambda x: {'id': x['mediaId'], 'url': f'{video_base_url}article/{x["mediaId"]}'},
+                'ranking': lambda x: {'id': x['mediaId'], 'url': f'{video_base_url}ranking/{x["mediaId"]}'}
```

Comment:
We only have extractors for post and stream, no?

Reply:
No extractors for articles and rankings.

Reply:
Then why are those results being returned?

Reply:
Because it's the honest thing to do. The user should know that the content exists, even though yt-dlp does not support it.

Reply:
No. Returning unsupported URL just raises an unnecessary error. Do rankings and articles even have video in them? Can you give an example?

Reply:
Articles and rankings do not have videos in them. What you call "unnecessary errors" are often quite informative and should not, in my opinion, be suppressed.

Reply:
yt-dlp is a **video**/audio downloader. It should not return pages without videos

Reply:
What if I add an extractor-specific option for this?

Reply:
Please explain to me why yt-dlp should return non-video URLs in playlists